util/src/Metrics.cpp

```diff
@@ -104,4 +104,84 @@ Counter Aggregator::GetCounter(const string& component_name,
                    component.values_.counters_);
 }
 
+// Generate a JSON string of all aggregated components. To save space we don't
+// add any newline characters.
+std::string Aggregator::ToJson() {
+  ostringstream oss;
+  std::lock_guard<std::mutex> lock(lock_);
+
+  // Add the object opening
+  oss << "{\"Components\":[";
+
+  // Add all the components
+  for (auto it = components_.begin(); it != components_.end(); ++it) {
+    // Add a comma after every component
+    if (it != components_.begin()) {
+      oss << ",";
+    }
+    oss << it->second.ToJson();
+  }
+
+  // Add the object end
+  oss << "]}";
+
+  return oss.str();
+}
+
+// Generate a JSON string of the component. To save space we don't add any
+// newline characters.
+std::string Component::ToJson() {
+  ostringstream oss;
+
+  // Add the object opening and component name
+  oss << "{\"Name\":\"" << name_ << "\",";
+
+  // Add any gauges
+  oss << "\"Gauges\":[";
+
+  for (int i = 0; i < names_.gauge_names_.size(); i++) {
+    if (i != 0) {
+      oss << ",";
+    }
+    oss << "{\"" << names_.gauge_names_[i] << "\":"
+      << values_.gauges_[i].Get() << "}";
```

Comment:
What's the reasoning behind a list of one-element maps, instead of just a map? That is, why `"Gauges":[{"alpha": 1},{"beta": 2},{"gamma":3}]` instead of `"Gauges":{"alpha":1,"beta":2,"gamma":3}`? Is it because the names might conflict? Or is this a format used by some other common tool?

Reply:
No reason. Will change.